mintapi/api.py

```diff
@@ -1744,29 +1748,48 @@ def initiate_account_refresh(email, password):
     return mint.initiate_account_refresh()
 
 
+def handle_password(type, prompt, email, password, use_keyring=False):
+    if use_keyring and not password:
+        # If we don't yet have a password, try prompting for it
+        password = keyring.get_password(type, email)
+
+    if not password:
+        # If we still don't have a password, prompt for it
+        password = getpass.getpass("Mint password: ")
+
+    if use_keyring:
+        # If keyring option is specified, save the password in the keyring
+        keyring.set_password(type, email, password)
+
+    return password
+
+
 def main():
     options = parse_arguments(sys.argv[1:])
 
     # Try to get the e-mail and password from the arguments
     email = options.email
     password = options.password
+    imap_account = options.imap_account
+    imap_password = options.imap_password
+    mfa_method = options.mfa_method
 
     if not email:
         # If the user did not provide an e-mail, prompt for it
         email = input("Mint e-mail: ")
 
-    if keyring and not password:
-        # If the keyring module is installed and we don't yet have
-        # a password, try prompting for it
-        password = keyring.get_password("mintapi", email)
-
-    if not password:
-        # If we still don't have a password, prompt for it
-        password = getpass.getpass("Mint password: ")
+    password = handle_password(
+        "mintapi", "Mint password: ", email, password, options.keyring
+    )
 
-    if options.keyring:
-        # If keyring option is specified, save the password in the keyring
-        keyring.set_password("mintapi", email, password)
+    if mfa_method == "email" and imap_account:
+        imap_password = handle_password(
+            "imap",
```

Comment:
Do you think it might be worth namespacing this to "mintapi_imap" or similar? If this just ends up in the keyring as "imap" that's a bit generic and could potentially collide with something else. Overall looks great!

Reply:
@mrooney that's a good point.  I'll change that namespace.